src/Tokenizer/Tokens.php

```diff
@@ -440,62 +440,11 @@ public function ensureWhitespaceAtIndex($index, $indexOffset, $whitespace)
      */
     public function findBlockEnd($type, $searchIndex, $findEnd = true)
     {
-        $blockEdgeDefinitions = self::getBlockEdgeDefinitions();
-
-        if (!isset($blockEdgeDefinitions[$type])) {
-            throw new \InvalidArgumentException(sprintf('Invalid param type: %s.', $type));
-        }
-
-        if (!self::isLegacyMode() && isset($this->blockEndCache[$searchIndex])) {
-            return $this->blockEndCache[$searchIndex];
-        }
-
-        $startEdge = $blockEdgeDefinitions[$type]['start'];
-        $endEdge = $blockEdgeDefinitions[$type]['end'];
-        $startIndex = $searchIndex;
-        $endIndex = $this->count() - 1;
-        $indexOffset = 1;
-
-        if (!$findEnd) {
-            list($startEdge, $endEdge) = [$endEdge, $startEdge];
-            $indexOffset = -1;
-            $endIndex = 0;
+        if (3 === func_num_args()) {
+            @trigger_error('Argument #3 of Tokens::findBlockEnd is deprecated and will be removed in 3.0, use Tokens::findBlockStart or drop the argument.', E_USER_DEPRECATED);
```

Comment:
I would be more specific.
if `$findEnd` is true then drop the argument, if false then use different methods.
we could trigger different error depends on the condition, so end user doesn't have to double-check what he shall do in his case.